images/base/files/usr/local/bin/entrypoint

```diff
@@ -29,7 +29,12 @@ fi
 
 grep_allow_nomatch() {
   # grep exits 0 on match, 1 on no match, 2 on error
-  grep "$@" || [[ $? == 1 ]]
+  grep "$@"|| [[ $? == 1 ]]
+}
+
+# regex_escape_ip converts IP address string $1 to a regex-escaped literal
+regex_escape_ip(){
+  sed -e 's#\.#\\.#g' -e 's#\[#\\[#g' -e 's#\]#\\]#g' <<<"$1"
```

Comment:
is this valid for ipv6?

Reply:
IPv6 has `[` potentially right? but no `.`
the colons shouldn't need escaping. but square brackets and periods are meaningful in regex and need escaping.

Reply:
we're just replacing regex-meaningful characters with the character prefixed by `\`, and only handling the subset we expect to appear in ip addresses, because that's a simpler problem than doing it for _all_ possible characters